terraso_backend/tests/graphql/mutations/test_soil_data.py

```diff
@@ -780,3 +781,253 @@ def test_apply_to_all(client, project_site, project_manager):
+                    },
+                },
+                # constraint violations
+                {
+                    "siteId": str(user_sites[1].id),
+                    "soilData": {
+                        "slopeAspect": -1,
+                        "depthDependentData": [],
+                        "depthIntervals": [],
+                        "deletedDepthIntervals": [],
+                    },
+                },
+                # no permission
+                {
+                    "siteId": str(non_user_site.id),
+                    "soilData": {
+                        "slopeAspect": 5,
+                        "depthDependentData": [],
+                        "depthIntervals": [],
+                        "deletedDepthIntervals": [],
+                    },
+                },
+                # does not exist
+                {
+                    "siteId": "00000000-0000-0000-0000-000000000000",
+                    "soilData": {
+                        "slopeAspect": 15,
+                        "depthDependentData": [],
+                        "depthIntervals": [],
+                        "deletedDepthIntervals": [],
+                    },
+                },
+            ]
+        },
+        client=client,
+    )
+
+    assert response.json()
+    assert "data" in response.json()
+    result = response.json()["data"]["pushSoilData"]
+    assert result["errors"] is None
+    assert result["results"][0]["result"]["soilData"]["slopeAspect"] == 10
+    assert result["results"][1]["result"]["reason"] == "INVALID_DATA"
+    assert result["results"][2]["result"]["reason"] == "NOT_ALLOWED"
+    assert result["results"][3]["result"]["reason"] == "DOES_NOT_EXIST"
+
+    user_sites[0].refresh_from_db()
+    assert user_sites[0].soil_data.slope_aspect == 10
+
+    user_sites[1].refresh_from_db()
+    assert not hasattr(user_sites[1], "soil_data")
+
+    non_user_site.refresh_from_db()
+    assert not hasattr(non_user_site, "soil_data")
+
+    history_0 = SoilDataHistory.objects.get(site=user_sites[0])
+    assert history_0.update_failure_reason is None
+    assert history_0.update_succeeded
+    assert history_0.soil_data_changes["slope_aspect"] == 10
+
+    history_1 = SoilDataHistory.objects.get(site=user_sites[1])
+    assert history_1.update_failure_reason == "INVALID_DATA"
+    assert not history_1.update_succeeded
+    assert history_1.soil_data_changes["slope_aspect"] == -1
+
+    history_2 = SoilDataHistory.objects.get(site=non_user_site)
+    assert history_2.update_failure_reason == "NOT_ALLOWED"
+    assert not history_2.update_succeeded
+    assert history_2.soil_data_changes["slope_aspect"] == 5
+
+    history_3 = SoilDataHistory.objects.get(site=None)
+    assert history_3.update_failure_reason == "DOES_NOT_EXIST"
+    assert not history_3.update_succeeded
+    assert history_3.soil_data_changes["slope_aspect"] == 15
+
+
+def test_push_soil_data_success(client, user):
```

Comment:
suggestion: from a readability perspective it might be easier to have the success case before the mixed case?